### DISCUSSION TITLE
label :wrap not working?

### DISCUSSION THREAD

**Discussion by LurkAndLoiter** [elkowar/eww#1149] [Q&A] [answered]
I'm attempting to write a notification sub-window and I'm having trouble getting :wrap to engage on labels. I expect the Ipsum Lorum to wrap into new lines once it hits the width instead it just truncates ... I have a feeling I'm missing something but for the life of me my troubleshooting has lead me nowhere. Hoping a new pair of eyes might see the issue.

![20240808_13h53m35s_grim](https://github.com/user-attachments/assets/14b46480-40ea-4009-88e7-82164c91d213)

``` yuck
(defwindow notificationPanel
  :monitor 1 
  :stacking "fg"
  :windowtype "normal"
  :wm-ignore true
  :geometry (geometry
              :width 300
              :height 300
              :halign "center"
              :spacing 5
              :anchor "top right")
  (box
    :class "popout-window"
    (_notifications)))
```

```
; TODO replace with deflisten
(defpoll notifications :interval "1m" `dunstctl history | jq '.data[]'`)
```

``` yuck
(defwidget _notifications []
   (box 
     :class "notification-widget"
     :space-evenly false
     :orientation "v"
     (label
       :text "${notifications}"
       :visible false)
     (button :onclick "dunstctl history-clear" "Clear All")
     (for notification in notifications
       (box
         :class "notification-box"
         :orientation "v"
         (box 
           :hexpand true
           :space-evenly false
           (label
             :xalign 0
             :hexpand true
             :text "${notification.summary.data}")
           (button
             :halign "end"
             :onclick "dunstctl history-rm ${notification.id.data}" "X"))
         (box 
           (label
             :wrap true
             :xaliign 0
             :text "Officiis odio harum officia accusamus corrupti hic fugiat."))))))
```

@nfragment Requested simplified version.

![20240812_02h31m19s_grim](https://github.com/user-attachments/assets/a2cf957e-7eb0-4ab9-b9e9-b1cefd93783e)

```
(defwindow testing
  :monitor 1 
  :stacking "fg"
  :windowtype "normal"
  :geometry (geometry
              :width 300
              :anchor "top right")
  (label :wrap true :text "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla porta dolor tincidunt metus lacinia convallis. Maecenas nulla nisi, vehicula quis quam vel, faucibus hendrerit magna.")) 
```

Answer:
For `:wrap true` to take effect on hyprland/wayland it appears the `:show-truncated false` also needs to be passed.

![20240812_12h17m52s_grim](https://github.com/user-attachments/assets/da3ad01c-dea8-437c-b963-b711e7b6dd0d)

```
(defwindow testing
  :monitor 1 
  :stacking "fg"
  :geometry (geometry
              :width 300
              :anchor "top right")
  (label :show-truncated false :wrap true :text "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla porta dolor tincidunt metus lacinia convallis. Maecenas nulla nisi, vehicula quis quam vel, faucibus hendrerit magna.")) 
```

![20240812_12h18m19s_grim](https://github.com/user-attachments/assets/bbb247b4-b672-4dff-952f-8459ee7ac3d5)

**Comment by nfragment** [comment]
Please try testing it without styling/css/scss. Or post the styling here.

**Reply by LurkAndLoiter** [reply]
Added screenshot and codeblock to original posting. Seems to just be general label issue I'm having there is no scss attached to testing. Edit I did confirm the text acts as this with a fully blank eww.scss since it was inheriting some general scss but even with the blank scss it ends up truncated as shown and not wrapped

**Reply by nfragment** [reply]
Looks like the width property is not having effect. 
what wm/de are you using ?
Do you want it as a "normal window" ? If not Try using other windowtype and `:wm-ignore true`

**Reply by LurkAndLoiter** [reply, marked as answer]
Trying all the windowtypes available per the wiki and :wm-ignore t/f resulted in no visible changes. wm : hyprland.
I did get wrapping to work. See original post for block and screenshots. It seems you need to disable truncation for wrap to have any effect. Thanks for the effort. I'm curious if other wm/de setups do not require `:show-truncated false` ?

**Reply by nfragment** [reply]
Glad you got it working and i got to learn something too :+1: 

I am on x11 dwm and does not require it

Here is all that is needed to get wrapping label.

```
(defwindow testing
    :monitor 0 
    :stacking "fg"
    :windowtype "dock"
    :wm-ignore true
    :geometry (geometry
                :width 300
                :anchor "top right")
    (label :wrap true :halign "start" :text "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla porta dolor tincidunt metus lacinia convallis. Maecenas nulla nisi, vehicula quis quam vel, faucibus hendrerit magna."))    
```

May be EWW has slightly different defaults for wayland.
